### ISSUE TITLE
Design for a single conflict resolution

### ISSUE THREAD

**Issue by syncush** [MapColonies/conflicts-ui#5]
We need to design how should one can view a conflict on a map (geometry) and the entity data (tags).

1. How should we display the geometry?
2. How should we show the conflict tags conflict?

would it be git like? showing two `osmChanges.json` and then an `outputChange` as a result?

just throwing an idea here, what do you think?

NOTE: I am not willing that the user will learn the osmChange.xml format, I posted the `gif` and `png` so we can get an idea of the layout

Gif showing the merging:

https://support.gitkraken.com/img/documentation/repositories/local_repository/merge-tool-toggle.gif

![image](https://user-images.githubusercontent.com/19842756/89156590-9ec6b780-d573-11ea-8274-1f55d1af7a87.png)

**Comment by rannyeli** [comment]
I think we should display both geometry versions using different colors on the map, and an option two choose between them.
About the tags, your idea sounds good, we can use some more easy and intuitive diff like in material-ui:
![image](https://user-images.githubusercontent.com/42379604/89159606-6675a800-d578-11ea-8da6-b169a92b2624.png)

**Comment by syncush** [comment]
We would basically need 3 panels, the same layout as gitkraken, we can accept a tag from entity A and another tag from entity B

@CptSchnitz what do you think?

**Comment by CptSchnitz** [comment]
I think ranny's idea is good, with a look and feel that is less "git like", but serve the same purpose, maybe with selecting instead of moving.
about the tag's themselves, do we also allow editing/removing of tags, or just to choose between the two options?

**Comment by syncush** [comment]
We would like as flexible as possible, so we should have a `feature toggle` for allowing editing/removing of tags.

**Comment by CptSchnitz** [comment]
would we also want something like "review before submitting"?

**Comment by syncush** [comment]
Do you mean that `User A` which is the under `User B` asks for a review and then `User B` has to confirm it before marking the conflict as resolved?

**Comment by CptSchnitz** [comment]
no, just a user before submitting can review the data in a simplified view, to check for any errors.

**Comment by syncush** [comment]
yeah I think it is important, will it be a modal ? or an `output` panel just above the `submit` buttom

**Comment by CptSchnitz** [comment]
I think modal is better so all the focus of the user is on that, and it's not something you just scroll past.

**Comment by CptSchnitz** [comment]
For now to test the ideas for the components tag we will create a POC based on the following sketch:
![image](https://user-images.githubusercontent.com/12687466/90010683-ee5e5f00-dca8-11ea-8f5a-c64d76383c03.png)

**Comment by CptSchnitz** [comment]
This gif shows the result of the POC, and we will continue with this design and integrate it into the app.
![tag conflict](https://user-images.githubusercontent.com/12687466/90155803-69e90a80-dd94-11ea-90a6-3c0cb7d0c2a5.gif)

**Comment by syncush** [comment]
Regarding the geometry, how should we display it? 
I thought of having 2 vector layers with a single feature in each. they have two distinct colors, when blent, the overlap is clear and visible (using opacity for both geometries).

**Comment by CptSchnitz** [comment]
some questions about the geometry display:
1. do we want to limit the extent of the map (e.g prevent moving out of the area)?
2. will parts missing from one geometry or the other will be clear enough when using only 2 colors?
3. which data we want to show when clicking on a feature? (if we want at all).

**Comment by syncush** [comment]
1. Yes, the extent should be reasonable.
2. When picking the 2 colors we have to choose 2 that blend well.
3. Metadata + tags?
